app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -170,9 +177,32 @@ def process_collection(collection, key)
         end
       end
 
-      def gather_data_for_this_region(arm_service, method = "list")
-        @data[:resource_groups].collect do |resource_group|
-          arm_service.send(method, resource_group[:name])
+      # Gather all data for all resource groups. If a subscription ID is provided,
+      # then only collect data for that subscription. Otherwise, collect all data
+      # for all subscriptions.
+      #
+      def gather_data_for_this_region(arm_service, method = 'list_all', subscription_id = nil)
+        subscriptions = subscription_id ? [subscription_id] : @subscriptions
+
+        subscriptions.collect do |sub_id|
+          arm_service.configuration.subscription_id = sub_id
+
+          begin
+            if method.to_s == 'list_all'
+              arm_service.send(method).select { |obj| obj.location == @ems.provider_region }
+            else
+              @data[:resource_groups].collect do |resource_group|
+                arm_service.send(method, resource_group[:name])
+              end
+            end
+          rescue Azure::Armrest::UnauthorizedException => err
+            if err.code == 'InvalidAuthenticationTokenTenant'
+              _log.warn("Invalid subscription for this tenant. Skipping.")
+              next
+            else
+              raise
+            end
+          end
```

Comment:
@djberg96 
I am seeing an exception with error code `AuthorizationFailed` when the `list_all` method is called. The message is:
`The client 'my_client' with object id 'object_id' does not have authorization to perform action 'Microsoft.Compute/virtualMachines/read' over scope '/subscriptions/my_subscription_id'."}} (Azure::Armrest::ApiException)`